### DISCUSSION TITLE
DIDComm v2

### DISCUSSION THREAD

**Discussion by TimoGlastra** [openwallet-foundation/credo-ts#1539] [Architecture & Design]
Discussion for remaining tasks and coordinating the remaining work.

Involved in the work: @Artemkaaas @genaris

Please add your Github handle here / add a comment to inform others you are interested in working on this

**Comment by Artemkaaas** [comment]
Remaining work:

-  Protocols adoption: issue-credential / present-proof / etc
  - Currently, you can only send `ping` message 
- DIDComm Signed messages - [spec](https://identity.foundation/didcomm-messaging/spec/#didcomm-signed-messages)
  - Not implemented as there was a need
- Supported key types:
  - Usage of `P-384` and `P-521` keys is not currently supported (aries-askar limitation)
  - `P-521` is optional according to the spec but `P-384` is mandatory: [spec](https://identity.foundation/didcomm-messaging/spec/#curves-and-content-encryption-algorithms) 
- Mediator support: routing and message pickup protocols

**Comment by berendsliedrecht** [comment]
`p-384` is added to askar (https://github.com/hyperledger/aries-askar/pull/157) but I am not sure if this is added in the JS wrapper as well.
